### PR TITLE
fix(geometry): guard denom == 0 in depth_from_plane_equation to prevent infinite depth

### LINKED ISSUE #4280: W1b: `depth_from_plane_equation` — the singularity guard is a no-op exactly at the singularity
`depth_from_plane_equation` guards its division by the ray-plane dot product with

```python
denom = torch.where(mask, eps * torch.sign(denom), denom)     # mask = denom.abs() < eps
```

`torch.sign(0.) == 0.`, so at `denom == 0` — the exact singularity the guard exists for — the epsilon is multiplied by zero and the guard is a no-op. A ray exactly parallel to the plane returns `inf`.

The masked-guard *shape* is right (unlike an `x + eps` that perturbs every result, this one only touches the masked branch — and it works: a denominator of `2.4e-09`, inside the mask but non-zero, is clamped to `±eps` and returns `±2e8`). Only the worst case is missed.

A grazing ray is not an exotic input: it is every pixel on the horizon of a ground plane.

## Repro

```python
import torch
from kornia.geometry.depth import depth_from_plane_equation

K = torch.tensor([[100.0, 0.0, 4.0], [0.0, 100.0, 3.0], [0.0, 0.0, 1.0]])[None]
px = torch.tensor([[[4.0, 3.0]]])            # the principal point: its ray is (0, 0, 1)
# a plane whose normal is exactly perpendicular to that ray -> n . ray == 0 exactly
print("grazing ray, denom == 0 exactly :", depth_from_plane_equation(torch.tensor([[0.0, 1.0, 0.0]]), torch.tensor([[2.0]]), px, K).item())
print("torch.sign(0.0)                 :", torch.sign(torch.tensor(0.0)).item(), "<- the guard multiplies eps by this")
# a denominator just inside the mask, but non-zero: the guard does clamp it
n = torch.tensor([[0.0, 1.0, 2.384185791015625e-09]])
print("denom in (0, eps)               :", depth_from_plane_equation(n, torch.tensor([[2.0]]), px, K).item())
print("denom in (-eps, 0)              :", depth_from_plane_equation(-n, torch.tensor([[2.0]]), px, K).item())
```

## Observed

```
grazing ray, denom == 0 exactly : inf
torch.sign(0.0)                 : 0.0 <- the guard multiplies eps by this
denom in (0, eps)               : 200000000.0
denom in (-eps, 0)              : -200000000.0
```

## Expected

The sign is taken with a zero-safe fallback, so `denom == 0` is clamped like every other point inside the mask:

```python
sign = torch.where(denom < 0, -1.0, 1.0)      # or torch.sign(denom) + (denom == 0)
denom = torch.where(mask, eps * sign, denom)
```

Nothing outside `|denom| < eps` changes, so every currently finite result is byte-identical; only the `inf` becomes a large finite depth. That makes it a focused fix, welcome as a PR, with a regression test that asserts the grazing ray is finite.

The alternative reading — that a grazing ray genuinely has no depth and `inf` is the right answer — is coherent too, but then the mask should be removed rather than left half-working, because today the function returns `inf` at `denom == 0` and `2e8` at `denom = 2.4e-09`, which is not a policy.

## Classification

**Broken output** — a singularity guard that does not fire at the singularity. Idiom 1 (an epsilon that guards a singularity, applied so that the singular case escapes it).

## Disposition

n/a — single symbol.

## Environment

kornia 0.9.0rc1 @ `92e5fe8d` (main), torch 2.14.0, python 3.11, cpu. Found during the camera conventions audit (batch 5, tracking issue #4262); the batch documents the behavior below as it is and pins it, and does not change it.

Posted on behalf of @ducha-aiki by Claude (Opus 5).

### PR DESCRIPTION
**Pull request by thegoodengineer**
## What does this pull request do?

\depth_from_plane_equation\ guarded its division by the ray-plane dot product with:
\\\python
denom_abs = torch.abs(denom)
zero_mask = denom_abs < eps
denom = torch.where(zero_mask, eps * torch.sign(denom), denom)
\\\
When \denom == 0\ (for instance, a grazing ray perpendicular to the plane normal), \	orch.sign(0.) == 0.\, multiplying \ps\ by zero and causing the guard to be a no-op. The division then produced \inf\.

This change uses a zero-safe fallback for \sign\ so that \denom == 0\ is clamped to \ps\ (or \-eps\ if negative), matching other values within \(-eps, eps)\ and ensuring the computed depth remains finite.

## Related issue or discussion

Closes #4280

## What did you check?

- Added a regression test \	est_grazing_ray_singularity\ in \	ests/geometry/test_depth.py\ checking that grazing rays with \denom == 0\ produce finite depth values matching \plane_offsets / eps\.
- Ran \pytest tests/geometry/test_depth.py::TestDepthFromPlaneEquation --device=cpu --dtype=float32,float64\ (all 19 tests pass).
- Ran \
uff check\ on modified files.

### CODE REVIEW

**Review by ducha-aiki — COMMENTED** [review]
This duplicates #4348: both close #4280, change the same denominator guard, and add the same grazing-ray regression; #4348 opened about five hours earlier. I recommend closing this PR and consolidating on #4348.

There is useful value to carry over: the zero-safe `torch.where` implementation at `kornia/geometry/depth.py:274` remains ONNX-exportable. I verified both legacy and dynamo ONNX export and ONNX Runtime output (`[[2e8]]`), whereas #4348's `torch.copysign` formulation breaks both exporters.

If this PR is retained instead, it is not mergeable yet: `tests/geometry/test_depth.py:544` fails the required CPU float16 job because `eps=1e-8` underflows to zero in float16 and `2 / 1e-8` cannot be represented finitely. The test/promise needs a representable numerical range or a dtype-aware policy, and `CHANGELOG.md:362` must cite this PR (`#4352`) rather than the issue.

Posted on behalf of [@ducha-aiki](https://github.com/ducha-aiki) by Codex (gpt-5.6-sol).

**Comment by thegoodengineer** [comment]
Closing this as superseded, as suggested. #4348 landed first and is on `main`, #4280 is closed, and its final form already uses the comparison-based `torch.where` guard rather than `torch.copysign`, so the ONNX exportability point is covered there too. The float16 test and CHANGELOG citation issues go away with the PR. Thanks for the review.
